src/utils/promise.utils.ts

```diff
@@ -4,16 +4,17 @@
  * outside of the promise constructor callback.
  *
  * @template T The type of the value with which the promise will be resolved.
+ * @template U The type of the reason with which the promise will be rejected.
  * @returns An object containing:
  * - `promise`: A Promise<T> that can be awaited or otherwise used.
  * - `resolve`: A function that, when called with a value of type T, resolves the promise.
- * - `reject`: A function that, when called with a reason, rejects the promise.
+ * - `reject`: A function that, when called with a reason of type U (defaults to unknown), rejects the promise.
  */
-export const createPromise = <T>() => {
-    let deferredResolve!: (value?: T | null) => void;
-    let deferredReject!: (reason?: unknown) => void;
+export const createPromise = <T, U = unknown>() => {
```

Comment:
For my understanding why we need this change? 🤔 

Cause AFAIR we are not pass argument to resolve and reject everywhere 🤔